packages/wp-story-editor/src/components/documentPane/status/status.js

```diff
@@ -242,16 +241,7 @@ function StatusPanel({
           break;
 
         case VISIBILITY.PRIVATE:
-          if (
-            !isUpdatedPublishModalEnabled &&
-            shouldReviewDialogBeSeen &&
-            !isAlreadyPublished
-          ) {
-            setShowReviewDialog(true);
-            properties.visibility = VISIBILITY.PRIVATE;
-          } else {
-            publishPrivately();
-          }
```

Comment:
@swissspidy  tagged you on this PR for this bit here, if anyone has intel on keeping this it's going to be you :) 

(copied from my PR description) 
```
Aside from some renaming, the only thing that's changing in this PR is the private publish flow. This was the only place that the review checklist dialog would show up. I checked on how posts work with the private flow, my update here has Story private publish working that same way. With the review checklist dialog you'd 1) select 'private' from the dropdown for visibility 2) get the default WP alert asking if you're sure, assuming you say yes 3) the review checklist dialog. Now you just get the first two steps.

I did this because if you're in the story details modal and you hit 'private' that's already 2 things popped up in your browser - the story details modal and the new alert asking if you're sure. It just feels excessive. I talked a while ago with Amy about it and this is just killing it officially.

If this isn't ok for some reason I don't know, we can put it back.
```

Reply:
Thanks for the ping!

This sounds reasonable to me.

Private stories are only visible for logged-in users. The review checklist/dialog is mostly relevant for public content though as we want to ensure eligibility for appearance on Search and Discover.

The one edge case here is when you want to make an existing _private_ story _public_. And in that case we show the dialog again, so that's all good!